src/main/java/Functionality.java

```diff
@@ -0,0 +1,93 @@
+import java.util.ArrayList;
+
+public class Functionality {
```

Comment:
I think Functionality can be quite general, what kind of functionality? Logic, storage, ui etc.